wri/gfw
# [FLAG-343] fire alert widget crashes page when clicking settings button

## Overview

Unlike most other widgets where the widget settings are predefined, in this widget (`firesAlertsSimple`) the min/max/start dates for the datepickers in the settings are determined based on the dates fetched from [an endpoint](https://data-api.globalforestwatch.org/dataset/nasa_viirs_fire_alerts/latest).

The `integratedDeforestationAlerts` widget also works in a similar fashion and suffers from the same bug; although it fetches data from a [glad endpoint](https://data-api.globalforestwatch.org/dataset/umd_glad_landsat_alerts/latest) instead. 

From what I learned during the debugging session, when the user opens the settings dropdown _before_ the dates are available to populate the datepickers, the datepickers crash (`undefined` dates). 

-----

I've considered a few options:  
1. Implement a mechanism to prevent the settings dropdown from opening until all data is available  
  _though this would mean re-engineering part of the core of the components that render the widgets in order to allow settings to be set asynchronously, which seems a bit overkill for just a couple outlier widgets_  
2. Hiding the settings button/dropdown while the widget is in a loading state  
  _however, this would cause the button to flicker when the widget is loading data to display in its body, such as when the user changes its settings_  
3. Hardcode dates in the widget's settings that the datepickers can use while the necessary data isn't returned by the endpoints
  _This seemed like a flakey solution and more of a patch; which I'm not sure how it'd influence the widget's data calculations_
4. Make the crashing datepickers themselves "fail gracefully".
  _in this case, if there aren't valid dates to be displayed yet, then the datepicker buttons show a loading spinner._

This PR implements option 4, as it seems like the easier and safer approach; any other widgets that happen to have the same bug and have been undetected should now not trigger the bug either. 

-----

**Notes:**  
- I've found a prop being passed that was unused [here](https://github.com/wri/gfw/pull/4703/commits/56297e41fed20a0fd5d1c48f309390dbc39bf5af), which I took the chance to remove in order to prevent confusion in the future  
- I've also noticed that the `WidgetHeader` component did expect a `loading` prop, which I've passed [here](https://github.com/wri/gfw/pull/4703/commits/33d01123cc8f04cfaad9bcd3ff020b8e0621ffea). It is used mainly to disable the dropdowns in the widget's settings while the widget is loading. 
- Other changes were just auto linting/prettier

## Testing

1. Visit [this link](https://gfw-staging-pr-4703.herokuapp.com/dashboards/aoi/64dbb15ca0f256001b1d8e92/?category=fires&location=WyJhb2kiLCI2NGRiYjE1Y2EwZjI1NjAwMWIxZDhlOTIiXQ%3D%3D&map=eyJjZW50ZXIiOnsibGF0IjotMTIuMjE2MDQ1MTIzODk5MDgsImxuZyI6LTQxLjUwOTgxMjc3OTAxMzU3fSwiem9vbSI6MTEuMzYwOTA2Njc1MjE2NDMsImJhc2VtYXAiOnsidmFsdWUiOiJwbGFuZXQiLCJjb2xvciI6IiIsIm5hbWUiOiJwbGFuZXRfbWVkcmVzX25vcm1hbGl6ZWRfYW5hbHl0aWNfMjAyMy0wN19tb3NhaWMifSwiY2FuQm91bmQiOmZhbHNlLCJkYXRhc2V0cyI6W3siZGF0YXNldCI6InBvbGl0aWNhbC1ib3VuZGFyaWVzIiwibGF5ZXJzIjpbImRpc3B1dGVkLXBvbGl0aWNhbC1ib3VuZGFyaWVzIiwicG9saXRpY2FsLWJvdW5kYXJpZXMiXSwiYm91bmRhcnkiOnRydWUsIm9wYWNpdHkiOjEsInZpc2liaWxpdHkiOnRydWV9LHsiZGF0YXNldCI6InRyZWUtY292ZXItbG9zcy1maXJlcyIsImxheWVycyI6WyJ0cmVlLWNvdmVyLWxvc3MtZmlyZXMiXSwib3BhY2l0eSI6MSwidmlzaWJpbGl0eSI6dHJ1ZSwicGFyYW1zIjp7InRocmVzaG9sZCI6MzAsInZpc2liaWxpdHkiOnRydWUsImFkbV9sZXZlbCI6ImFkbTAifX1dfQ%3D%3D&showMap=true)
6. Before the _"FIRE ALERTS IN AREA IN BAHIA, BRAZIL"_ widget finishes loading, click the _"Settings"_ button  
7. Verify that the Dashboard no longer crashes

## Tracking

[FLAG-343](https://gfw.atlassian.net/browse/FLAG-343)
